### ISSUE TITLE
Microphone might be turned off

### ISSUE THREAD

**Issue by captain-woof** [SimformSolutionsPvtLtd/audio_waveforms#73]
**Issue**:
Starting audio recording with `recorderController.record()` prints this:

```
D/AudioWaveforms( 8758): Microphone might be turned off
```

And nothing else happens. The waveforms don't appear.

**Version**:
audio_waveforms: 0.1.3

**Comment by Ujas-Majithiya** [comment]
If you are using an android emulator then it will be happening because sometimes emulators don't start microphones so I would suggest using a real device. 
Duplicate of #25.

**Comment by captain-woof** [comment]
But when I change the sample rate, set it to an explicit value of say 41000, then this error does not come up.

But I have a new error now, saying that my Row gets overflown (I have the audio wave and record button in a row). Why's this happening?

Here's the widget:

```dart
@override
  Widget build(BuildContext context) {
    return Container(
      margin: const EdgeInsets.symmetric(horizontal: 0, vertical: 12),
      child: Row(
        mainAxisAlignment: MainAxisAlignment.center,
        crossAxisAlignment: CrossAxisAlignment.center,
        children: [
          // Wave
          Expanded(
            child: AudioWaveforms(
                size: const Size(100, 80.0),
                recorderController: _recorderController,
                waveStyle: const WaveStyle(showDurationLabel: true)),
          ),

          // Pause/resume button (shown only when recording has started)
          IconButton(
            icon: Icon(
                _recorder2state == Recorder2State.paused
                    ? Icons.play_arrow
                    : Icons.pause,
                color: Colors.black),
            onPressed: _recorder2state != Recorder2State.beforeRecord
                ? togglePauseResume
                : null,
          ),

          // Start/stop button (shown always)
          IconButton(
            icon: Icon(
                _recorder2state == Recorder2State.beforeRecord
                    ? Icons.circle
                    : Icons.stop,
                color: Colors.red),
            onPressed: toggleStartStop,
          ),
        ],
      ),
    );
  }
```

**Comment by Ujas-Majithiya** [comment]
I have tried your provided sample code and it working fine for me. Overflow issue might be coming from some other widget. 
If you are still facing microphone issue for emulator then open emulator settings by pressing on three dots and tap on microphone and enable `virtual microphone uses host audio input`.

**Comment by captain-woof** [comment]
The overflow errors happens only after I start recording, and not before. It happens even if I don't use the `AudioWaveforms` widget.

**Comment by captain-woof** [comment]
Yeah, the error was happening coz of some other widget.

I still can't start recording through. I get this error:

```
PlatformException (PlatformException(error, start failed., null, java.lang.RuntimeException: start failed.
	at android.media.MediaRecorder.start(Native Method)
	at com.simform.audio_waveforms.AudioRecorder.startRecorder(AudioRecorder.kt:70)
	at com.simform.audio_waveforms.AudioWaveformsPlugin.onMethodCall(AudioWaveformsPlugin.kt:56)
	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage(MethodChannel.java:262)
	at io.flutter.embedding.engine.dart.DartMessenger.invokeHandler(DartMessenger.java:295)
	at io.flutter.embedding.engine.dart.DartMessenger.lambda$dispatchMessageToQueue$0$io-flutter-embedding-engine-dart-DartMessenger(DartMessenger.java:319)
	at io.flutter.embedding.engine.dart.DartMessenger$$ExternalSyntheticLambda0.run(Unknown Source:12)
	at android.os.Handler.handleCallback(Handler.java:873)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loop(Looper.java:193)
	at android.app.ActivityThread.main(ActivityThread.java:6669)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
))
```

**Comment by captain-woof** [comment]
I got it to working with 16000 khz, but, the audio quality is poor.

How to improve it?

**Comment by Ujas-Majithiya** [comment]
@captain-woof The quality of audio is dependent on underlying platform and device you are using. And can you please share how downgrading sample rate resolved your issue?

**Comment by captain-woof** [comment]
So, I was manually setting the sample rate to 41000, and that was failing.

Then, I switched to 16000, and it worked.

**Comment by ujas-m-simformsolutions** [comment]
Closing this because sample rate causing overflow issue doesn't line up and audio quality of the audio is dependent on platform and device, not the plugin.
